### PR TITLE
HW04 Pavel Karenda

### PR DESCRIPTION
**Pull request by ButterBrot777**
Solution for homework 04

### CODE REVIEW

**Comment by vladhilko** [comment]
Please, run `rubocop-rspec` 😔

File: HW04/Pavel_Karenda/spec/my_each_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  describe '#my_each' do
+    context 'when no block is given' do
+      it 'returns start array' do
+        expect(subject.my_each.to_a).to eq subject
+      end
+    end
+
+    context 'when block is given' do
+      it 'returns modified array' do
+        p 'asdf'
```
Comment (vladhilko) [inline HW04/Pavel_Karenda/spec/my_each_spec.rb:19]: remove it please

File: HW04/Pavel_Karenda/spec/my_map_spec.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+class Array
+  include CustomMap
+  include CustomEach
+end
+
+RSpec.describe Array do
+  subject { described_class.new([0, 1, 2, 3]) }
+
+  describe '#my_each' do
```
Comment (vladhilko) [inline HW04/Pavel_Karenda/spec/my_map_spec.rb:12]:
```suggestion
  describe '#my_map' do
```

File: HW04/Pavel_Karenda/spec/my_map_spec.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    context 'when no block is given' do
+      it 'returns start array' do
+        expect(subject.my_map.to_a).to eq subject
+      end
+    end
+
+    context 'when block is given' do
+      it 'returns modified array' do
+        expect(subject.my_map { |element| p element + 2 }).to eq([2, 3, 4, 5])
```
Comment (vladhilko) [inline HW04/Pavel_Karenda/spec/my_map_spec.rb:21]: Why do you need `p` here?
```suggestion
        expect(subject.my_map { |element| element + 2 }).to eq([2, 3, 4, 5])
```

File: HW04/Pavel_Karenda/spec/my_select_spec.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+class Array
+  include CustomSelect
+  include CustomEach
+end
+
+RSpec.describe Array do
+  subject { described_class.new([0, 1, 2, 3]) }
+
+  describe '#my_each' do
```
Comment (vladhilko) [inline HW04/Pavel_Karenda/spec/my_select_spec.rb:12]:
```suggestion
  describe '#my_select' do
```

File: HW04/Pavel_Karenda/spec/my_select_spec.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+    context 'when no block is given' do
+      it 'returns start array' do
+        expect(subject.my_select.to_a).to eq subject
+      end
+    end
+
+    context 'when block is given' do
+      it 'returns modified array' do
+        expect(subject.my_select { |element| p element != 0 }).to eq([1, 2, 3])
```
Comment (vladhilko) [inline HW04/Pavel_Karenda/spec/my_select_spec.rb:21]: the same

File: HW04/Pavel_Karenda/spec/my_each_spec.rb (hunk 1 above)
Comment (ButterBrot777) [reply HW04/Pavel_Karenda/spec/my_each_spec.rb:19]: sorry, it was debug element. Deleted.

File: HW04/Pavel_Karenda/spec/my_map_spec.rb (hunk 3 above)
Comment (ButterBrot777) [reply HW04/Pavel_Karenda/spec/my_map_spec.rb:21]: is was added to have something in stdout. Deleted.

File: HW04/Pavel_Karenda/spec/my_map_spec.rb (hunk 2 above)
Comment (ButterBrot777) [reply HW04/Pavel_Karenda/spec/my_map_spec.rb:12]: sorry, copypasta.

File: HW04/README.md [hunk 6]
```diff
@@ -1,18 +0,0 @@
-## HW04
-
-- use `Ruby >= 2.6`
-- please create fork from the https://github.com/labs-ruby/homeworks-2020 (should be
```
Comment (vladhilko) [inline HW04/README.md:4]: Can you return it back?

Comment (ButterBrot777) [reply HW04/README.md]: Done. Sorry for late :(

**Pull request merged**